alacritty/src/config/ui_config.rs

```diff
@@ -167,7 +167,13 @@ impl UiConfig {
     /// Derive [`PtyOptions`] from the config.
     pub fn pty_config(&self) -> PtyOptions {
         let shell = self.shell.clone().map(Into::into);
-        PtyOptions { shell, working_directory: self.working_directory.clone(), hold: false }
+        PtyOptions {
+            shell,
+            working_directory: self.working_directory.clone(),
+            hold: false,
+            #[cfg(not(windows))]
+            env: self.env.clone(),
```

Comment:
This should be _empty_ since we set env for alacritty iself so it's inherited.

The rest could be a bit confusing with the way things work currently.